it/repositories/application/FsbTestGroupMongoRepositorySpec.scala

```diff
@@ -0,0 +1,142 @@
+      val fsbTestGroup = FsbTestGroup(List(schemeEvaluationResult))
+      result mustBe fsbTestGroup
+    }
+
+    "return None if FsbTestGroup is not found for the given applicationId" in {
+      val applicationId = createApplication()
+      val result = repository.findByApplicationId("appId-with-no-fsb-test-group").futureValue
+      result mustBe None
+    }
+
+    "return None if the given applicationId does not exist" in {
+      repository.findByApplicationId("appId-that-does-not-exist").futureValue mustBe None
+    }
+  }
+
+  "findByApplicationIds" should {
+    "return the FsbSchemeResult for the given applicationIds" in {
+      val applicationId1 = createApplication()
+      val applicationId2 = createApplication()
+      val applicationId3 = createApplication()
+      val schemeEvaluationResult = SchemeEvaluationResult("GovernmentOperationalResearchService", Green.toString)
+
+      repository.save(applicationId1, schemeEvaluationResult).futureValue
+      repository.save(applicationId2, schemeEvaluationResult.copy(result = Red.toString)).futureValue
+      repository.save(applicationId3, schemeEvaluationResult).futureValue
+
+      val result = repository.findByApplicationIds(List(applicationId1, applicationId2, applicationId3), None).futureValue
+
+      val expectedResult = List(
+        FsbSchemeResult(applicationId1, List(schemeEvaluationResult)),
+        FsbSchemeResult(applicationId2, List(schemeEvaluationResult.copy(result = Red.toString))),
+        FsbSchemeResult(applicationId3, List(schemeEvaluationResult))
+      )
+
+      result must contain theSameElementsAs expectedResult
+    }
+
+    "return the FsbSchemeResult for the given applicationIds filtered by scheme" in {
+      val applicationId1 = createApplication()
+      val applicationId2 = createApplication()
+
+      repository.save(applicationId1, SchemeEvaluationResult("GovernmentOperationalResearchService", Red.toString)).futureValue
+      repository.save(applicationId1, SchemeEvaluationResult("GovernmentSocialResearchService", Green.toString)).futureValue
+      repository.save(applicationId2, SchemeEvaluationResult("GovernmentOperationalResearchService", Green.toString)).futureValue
+
+      val result = repository.findByApplicationIds(
+        List(applicationId1, applicationId2), Some(SchemeId("GovernmentOperationalResearchService"))).futureValue
+
+      val expectedResult = List(
+        FsbSchemeResult(applicationId1, List(SchemeEvaluationResult("GovernmentOperationalResearchService", "Red"))),
+        FsbSchemeResult(applicationId2, List(SchemeEvaluationResult("GovernmentOperationalResearchService", "Green")))
+      )
+
+      result must contain theSameElementsAs expectedResult
+    }
+
+  }
+
+  private def applicationRepository = new GeneralApplicationMongoRepository(DateTimeFactory, cubiksGatewayConfig)
+
+  private def createApplication(): String = {
+    val applicationId = generateUUID()
```

Comment:
Minor thing and you probably don't have to change but I prefer to invoke `UUIDFactory.generateUUID()` as opposed to mixing it in.